From d6d55912dce3f70588e5e938b7df0ee0806ca94c Mon Sep 17 00:00:00 2001
From: "Liu,Dalei" <daleix.liu@intel.com>
Date: Tue, 5 Jan 2016 19:01:46 +0800
Subject: [PATCH 3/4] atomisp: modify capt_pp_in_res for ISP2.7

ISP2.7 request capt_pp_in_res same to output info because
of not supporting downscale.

Change-Id: I8ed5728daaf4f656b1e0e9658e73b5989f9b254d
Original-Tracked-On: #H1504133721
Signed-off-by: Liu,Dalei <daleix.liu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/93470
Build: Dessert M BXT RVP (SC)
Build: Dessert M CHT FFD (SC)
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-15952
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/458630
---
 .../pci/atomisp2/atomisp_driver/atomisp_compat_css20.c   | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 1c1d193..1337d22 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -2639,10 +2639,18 @@ static void __configure_capture_pp_input(struct atomisp_sub_device *asd,
 
 	pipe_extra_configs->enable_yuv_ds = true;
 
-	pipe_configs->capt_pp_in_res.width =
-		stream_config->input_config.effective_res.width;
-	pipe_configs->capt_pp_in_res.height =
-		stream_config->input_config.effective_res.height;
+	/* ISP2.7 didn't support downscale */
+	if (pipe_configs->isp_pipe_version == SH_CSS_ISP_PIPE_VERSION_2_7) {
+		pipe_configs->capt_pp_in_res.width =
+			pipe_configs->output_info[0].res.width;
+		pipe_configs->capt_pp_in_res.height =
+			pipe_configs->output_info[0].res.height;
+	} else {
+		pipe_configs->capt_pp_in_res.width =
+			stream_config->input_config.effective_res.width;
+		pipe_configs->capt_pp_in_res.height =
+			stream_config->input_config.effective_res.height;
+	}
 
 	dev_dbg(isp->dev, "configuring pipe[%d]capture pp input w=%d.h=%d.\n",
 		pipe_id, width, height);
-- 
1.9.1

